controllers/registry_test.go

```diff
@@ -539,34 +535,30 @@ func TestBackOffRegistry_Get(t *testing.T) {
 					Name:      "bar",
 				}: {
 					bo: backoff.NewExponentialBackOff(
-						DefaultExponentialBackOffOpts()...,
+						backoff.WithRandomizationFactor(0.1),
 					),
 				},
 			},
 			objKey: client.ObjectKey{
 				Namespace: "foo",
 				Name:      "bar",
 			},
-			want: &BackOff{
-				bo: backoff.NewExponentialBackOff(
-					DefaultExponentialBackOffOpts()...,
-				),
-			},
-			want1: false,
+			want: false,
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
+			tt := tt
+			t.Parallel()
 			r := &BackOffRegistry{
 				m:    tt.m,
 				opts: tt.opts,
 			}
-			got, got1 := r.Get(tt.objKey)
-			assert.NotNilf(t, got, "Get(%v)", tt.objKey)
-			assert.Equalf(t, tt.want1, got1, "Get(%v)", tt.objKey)
-			last := got.bo.NextBackOff()
-			assert.Greaterf(t, last, time.Duration(0), "Get(%v)", tt.objKey)
-			assert.Greaterf(t, got.bo.NextBackOff(), last, "Get(%v)", tt.objKey)
+			bo, created := r.Get(tt.objKey)
+			assert.NotNilf(t, bo, "Get(%v)", tt.objKey)
+			assert.Equalf(t, tt.want, created, "Get(%v)", tt.objKey)
+			assert.Greaterf(t, bo.NextBackOff(), time.Duration(0), "Get(%v)", tt.objKey)
+			assert.Lessf(t, bo.NextBackOff(), bo.NextBackOff(), "Get(%v)", tt.objKey)
```

Comment:
Does this match how the backoff registry is being used? I thought there would be an `r.Get()` in-between calls to `bo.NextBackoff()`.

Reply:
Great call out. Updated in cc31c2dc645af549833c5d5314fc2f8eed63a144